packages/flutter_tools/lib/src/run_cold.dart

```diff
@@ -205,35 +210,13 @@ class ColdRunner extends ResidentRunner {
     if (details) {
       printHelpDetails();
     }
-    commandHelp.h.print();
+    commandHelp.h.print(); // TODO(ianh): print different message if details is false
     if (_didAttach) {
       commandHelp.d.print();
     }
     commandHelp.c.print();
     commandHelp.q.print();
-    for (final FlutterDevice device in flutterDevices) {
-      final String dname = device.device.name;
-      if (device.vmService != null) {
-        // Caution: This log line is parsed by device lab tests.
-        globals.printStatus(
-          'An Observatory debugger and profiler on $dname is available at: '
-          '${device.vmService.httpAddress}',
-        );
-
-        final DevToolsServerAddress devToolsServerAddress = activeDevToolsServer();
-        if (devToolsServerAddress != null) {
-          final Uri uri = devToolsServerAddress.uri?.replace(
-            queryParameters: <String, dynamic>{'uri': '${device.vmService.httpAddress}'},
-          );
-          if (uri != null) {
-            globals.printStatus(
-              '\nFlutter DevTools, a Flutter debugger and profiler, on '
-              '${device.device.name} is available at: $uri',
-            );
-          }
-        }
-      }
-    }
+    printDebuggerList();
```

Comment:
if this method sets `_reportedDebuggers` to true, then won't we be in a race for whether DevTools has successfully served by the time this is launched? If `_reportedDebuggers` is true by the time `serveAndAnnounceDevTools` is called, `printDebuggerList` won't be called from `serveAndAnnounceDevTools`. 

And on the flip side, if `printDebuggerList` is called from `serveAndAnnounceDevTools` before this line is hit, won't we be printing the debugger list twice?

Reply:
There's two paths:
1. DevTools launches before we first call printDebuggerList as part of the first help message. In that case, DevTools launching is silent (it doesn't call printDebuggerList because the flag is false), and printDebuggerList lists DevTools as a debugger.
2. DevTools launches after we first call printDebuggerList. We first print the help message without DevTools, then set the flag; when DevTools finishes launching, it sees the flag is set and calls printDebuggerList to list the DevTools debugger.

Assuming the user does nothing, the output is identical in both cases. (Well, it's slightly different in the case where we're launching on multiple devices, but only insofar as the Observatories are listed first then the DevTools in case #2 instead of them being interlaced as in case #1, but I think we can live with that!)